ci/test-stable.sh

```diff
@@ -69,7 +69,13 @@ test-stable-bpf)
       "$cargo_test_bpf" --bpf-sdk ../../../../sdk/bpf
       popd
     fi
-  done
+  done |& tee cargo.log
+  solana_program_count=$(grep -c 'solana-program v' cargo.log)
+  rm -f cargo.log
+  if ((solana_program_count > 4)); then
+      echo "Regression of build redundancy. Review dependency features that trigger redundant rebuilds of solana-program."
+      exit 1
+  fi
```

Comment:
I don't exactly follow how this works. What input will be echoed into `cargo.log`? Why 4 was chosen as the max count for solana program? Can you add some comments to explain?

Reply:
The input is output of running cargo which includes names of all crates being compiled. We check the number of solana-program crate compilations. When the first of solana/programs/bpf/rust/* crates is built, solana-program crate is compiled 3 times, once for different targets. An additional 4th compilation of solana-program crate happens when solana/program/rust/bpf/simulation crate is being built. However, the 4th compilation is of a different version of solana-program. If features are not consolidated, additional solana-program compilations will be performed when other solana/programs/bpf/rust/* crates are being built. I can put this information as a comment in the CI script.

Reply:
Ah ok, makes sense, thanks for the explanation. An extra comment in the script sounds great, thanks!